build/main_test.go

```diff
@@ -34,3 +34,45 @@ func ExampleTransactionBuilder() {
 	fmt.Printf("tx base64: %s", txeB64)
 	// Output: tx base64: AAAAADZY/nWY0gx6beMpf4S8Ur0qHsjA8fbFtBzBx1cbQzHwAAAAZAAAAAAAAAABAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAALSRpLtCLv2eboZlEiHDSGR6Hb+zZL92fbSdNpObeE0EAAAAAAAAAAB3NZQAAAAAAAAAAARtDMfAAAABA2oIeQxoJl53RMRWFeLB865zcky39f2gf2PmUubCuJYccEePRSrTC8QQrMOgGwD8a6oe8dgltvezdDsmmXBPyBw==
 }
+
+// ExamplePathPayment creates and signs a simple transaction with PathPayment operation, and then
+// encodes it into a base64 string capable of being submitted to stellar-core.
+func ExamplePathPayment() {
```

Comment:
Notice that there is no separate `PathPayment` mutator.  Instead, I use the normal `Payment` mutator and it knows that when I start provided path payment specific mutators that it should build a PathPaymentOp instead.  This allows the end developer to not care whether to pick one or the other.  The system will just use the most appropriate operation.

IMO, this is consistent with how we expose non-native payments in the higher level SDKs...  We paint over some of the specifics to give a simpler external interface.  For example, js-sdk's `Asset` doesn't differentiate between alphanum4 and alphanum12 assets... it will just transparently use the correct one based upon the code supplied.